compiler/optimizer/VPHandlers.cpp

```diff
@@ -2842,6 +2878,15 @@ TR::Node *constrainIaload(OMR::ValuePropagation *vp, TR::Node *node)
             }
          }
       }
+
+   if (sig)
+      {
+      constraint = createTypeHintConstraint(vp, node, sig, len);
+      if (constraint)
+         {
+         vp->addGlobalConstraint(node, constraint);
+         }
+      }
```

Comment:
1. > I think it would make sense for this to happen earlier in the handler, where we normally deal with field signatures

Do you mean moving `createTypeHintConstraint` to after the`if (classBlock)` check in [1] ?

2. > And again we should only need to do it when classBlock is null

Similar question as the above, the existing code creates `VPClassType` only if `classBlock && jlClass && classBlock != jlClass`. Don't we want to create hint in other cases when `classBlock` is non-NULL? 

[1]
https://github.com/eclipse/omr/blob/6bbfdb51b9b5b16eb6c3fe98ffe073d0a2b4efa0/compiler/optimizer/VPHandlers.cpp#L2680-L2704

Reply:
Please see the answers to the corresponding questions in `constrainAload()`

Reply:
Addressed in bb6ab93